src/libaktualizr/primary/aktualizr.h

```diff
@@ -26,18 +26,57 @@ class Aktualizr {
    * Launch aktualizr. Depending on the \ref RunningMode in the configuration,
    * this may run indefinitely, so you may want to run this on its own thread.
    */
-  int run();
+  int Run();
+
+  /**
+   * Asynchronously shutdown Aktualizr
+   */
+  void Shutdown();
+
   /**
-   * Send a command to the SotaUptaneClient command interpreter.
-   * @param command the command to send.
+   * Asynchronously perform a check for campaigns.
+   * Campaigns are a concept outside of Uptane, and allow for user approval of
+   * updates before the contents of the update are known.
    */
-  void sendCommand(const std::shared_ptr<command::BaseCommand>& command);
+  void CampaignCheck();
+
+  /**
+   * Asynchronously send local device data to the server.
+   * This includes network status, installed packages, hardware etc.
+   */
+  void SendDeviceData();
+
+  /**
+   * Asynchronously fetch Uptane metadata.
+   * This collects a client manifest, PUTs it to the director, then updates
+   * the Uptane metadata, including root and targets.
+   */
+  void FetchMetadata();
+
+  /**
+   * Asynchronously load already-fetched Uptane metadata from disk.
+   * This is only needed when the metadata fetch and downloads/installation are
+   * in separate aktualizr runs.
+   */
+  void CheckUpdates();
```

Comment:
Is this accurate or necessary? Previously, `CheckUpdates` was internal only, since it was automatically generated after a `FetchMetaComplete`. We could expose it if you want it, but is there any advantage to just stuffing all of that functionality inside `FetchMeta`, other than that `CheckUpdates` doesn't require network connectivity? I fear unnecessary expansion of the API to support a trivial advantage in functionality.

Reply:
I was trying to avoid exposing it, but `CheckUpdates` is used in main.cc:

~~~
    switch (config.uptane.running_mode) {
      case RunningMode::kDownload:
      case RunningMode::kInstall:
        aktualizr.CheckUpdates();
        break;
~~~

It is a somewhat special case, because it initializes the list of targets to be downloaded/installed from the Uptane metadata on disk.  I'm not sure how be can avoid that, given the requirement for a 'download' and 'install' that don't fetch metadata.

Reply:
Is that a requirement? If we have to do a fetch at some point, and the fetch eventually creates a `CheckUpdates` internally, then why not just do that? If we reboot, what is the problem with fetching again before allowing a download? The download requires connectivity anyway. If we reboot between downloading and installing, I'll grant that that is trickier. Can we do a `CheckUpdates` before installation if the metadata is marked as unchecked?

Reply:
I think the requirement was that at least the 'install' step could be done without internet connectivity.

Reply:
So for now, maybe this is fine, but could we need detect if the metadata is unchecked since last reboot and then automatically do a `CheckUpdates` if we need it? This seems like something the user should not have to care about at all.